src/button/src/components/Button.js

```diff
@@ -20,66 +21,58 @@ type Props = {
   children :Node;
   className ? :string;
   disabled ? :boolean;
+  isLoading ? :boolean;
   mode ? :ButtonMode;
-  onClick :() => void;
+  onClick :(e :SyntheticEvent<HTMLButtonElement>) => void;
   type ? :string;
 };
 
-type State = {};
-
 /*
  * Inspiration:
  * https://atlaskit.atlassian.com/packages/core/button
  * https://evergreen.surge.sh/components/buttons
  */
-export default class Button extends Component<Props, State> {
+const Button = (props :Props) => {
```

Comment:
It was not choice made for any real engineering purpose other than to keep the linter happy `react/prefer-stateless-function` since Button did not have any methods or state to manage for itself.